### PR TITLE
Tests refactor

### PR DESCRIPTION
**Pull request by marcharper**
Even more tests and I reworked several more of the test files while attempting to adhere to PEP8.

### CODE REVIEW

**Comment by meatballs** [comment]
I like the look of this a lot!! Probably should be somebody other than me that merges it now though as it has my sticky paw marks on it.

File: axelrod/tests/test_grudger.py [hunk 1]
```diff
@@ -111,21 +103,13 @@ class TestOppositeGrudger(TestPlayer):
 
     def test_initial_strategy(self):
         """
-        Starts by cooperating
+        Starts by cdefecting
```
Comment (drvinceknight) [inline axelrod/tests/test_grudger.py]: This is a great PR. I noticed this one spelling mistake. Happy to merge after.

**Comment by marcharper** [comment]
Sorry about that -- I had some more commits and had to use some git-fu to move them around. Spelling error fixed!

**Comment by drvinceknight** [comment]
One other thing that I've just thought of is the documentation: http://axelrod.readthedocs.org/en/latest/contributing.html#how-to-write-tests

that probably needs to be changed now. If it's too much trouble to tweak it, I'm happy to do it, and then one of you can merge in to say you're happy with what I've written (that most probably won't happen today though).

**Comment by marcharper** [comment]
Well one can still write tests that way so at least it's not wrong. I took a quick stab at explaining the new tests.

**Comment by drvinceknight** [comment]
:+1: Awesome PR :)

**Comment by meatballs** [comment]
This has re-introduced a bug on line 39 of tournament.py

**Comment by meatballs** [comment]
I bow to the wisdom Mr Einstein in these circumstances:

> Anyone who has never made a mistake has never tried anything new.

**Comment by drvinceknight** [comment]
Damn we really need #156 ...

**Comment by drvinceknight** [comment]
I meant #172 ...

**Comment by meatballs** [comment]
Yep - I only caught it when I tried to merge master into my local copy of branch 166.

**Pull request merged**